doc/source/qibo.rst

```diff
@@ -156,9 +181,16 @@ all the gates in the group.
 Time evolution
 ^^^^^^^^^^^^^^
 
+State evolution
+"""""""""""""""
```

Comment:
I would move the "Time evolution" subsection above the "Circuit addition" and "Circuit fusion" sections because I think it would look better in the side contents list.

Also, since now we list each model as a seperate section in the contents, do you think we should still keep the general / application specific circuit model distinction or shall we put all circuits in the same section and just have one main section for circuit-based computation and one for time evolution?